pandas/indexes/range.py

```diff
@@ -46,20 +47,17 @@ def __new__(cls, start=None, stop=None, step=None, name=None, dtype=None,
 
         # RangeIndex
         if isinstance(start, RangeIndex):
-            if not copy:
-                return start
```

Comment:
the issue is that ``.view`` needs to be fixed instead. It should do a ``_shallow_copy`` rather than returning the same instance